cloudigrade/account/tasks.py

```diff
@@ -143,14 +146,84 @@ def scale_up_inspection_cluster():
 
 @retriable_shared_task
 @rewrap_aws_errors
-def run_inspection_cluster(ami_volume_list):
+def run_inspection_cluster(messages, cloud='aws'):
     """
     Run task definition for "houndigrade" on the cluster.
 
-    Todo:
-        - Implement this!
+    Args:
+        messages (list): A list of dictionary items containing
+            the ami_id and volume_id to be inspected.
+        cloud (str): String key representing what cloud we're inspecting.
 
     Returns:
         None: Run as an asynchronous Celery task.
 
     """
+    task_command = ['-c', cloud]
+    if settings.HOUNDIGRADE_DEBUG:
+        task_command.extend(['--debug'])
+
+    # get ecs container instance id
+    ecs = boto3.client('ecs')
+    result = ecs.list_container_instances(
+        cluster=settings.HOUNDIGRADE_ECS_CLUSTER_NAME)
+
+    # verify we have our single container instance
+    num_instances = len(result['containerInstanceArns'])
+    if num_instances == 0:
+        raise AwsECSInstanceNotReady
+    elif num_instances > 1:
+        raise AwsTooManyECSInstances
+
+    result = ecs.describe_container_instances(
+        containerInstances=[result['containerInstanceArns'][0]],
+        cluster=settings.HOUNDIGRADE_ECS_CLUSTER_NAME
+    )
+    instance_id = result['containerInstances'][0]['ec2InstanceId']
+
+    # attach volumes
+    ec2 = boto3.resource('ec2')
+    for index, message in enumerate(messages):
+        mount_point = settings.HOUNDIGRADE_MOUNT_POINTS[index]
```

Comment:
Although we _should_ have `HOUNDIGRADE_AWS_VOLUME_BATCH_SIZE` equal to or less than the length of `HOUNDIGRADE_MOUNT_POINTS`, if either of those is misconfigured, this could result in an `IndexError`. How would you feel about checking that explicitly? Or maybe in the settings module itself do a quick sanity-check the those variables and blow up at startup if the batch size is ever greater than the number of mount points?

Reply:
Now that we no longer have a list, and generate the names on demand, I am resorting to trusting the `scale_up_inspection_cluster` to only give me enough messages up to the batch limit.